app/src/common/gecko/com/igalia/wolvic/browser/api/impl/SessionImpl.java

```diff
@@ -0,0 +1,391 @@
+            return;
+        }
+        mHistoryDelegate = delegate;
+        if (mHistoryDelegate == null) {
+            mSession.setHistoryDelegate(null);
+        } else {
+            mSession.setHistoryDelegate(new HistoryDelegateImpl(mHistoryDelegate, this));
+        }
+    }
+
+    @Nullable
+    @Override
+    public HistoryDelegate getHistoryDelegate() {
+        return mHistoryDelegate;
+    }
+
+    @Override
+    public void setContentBlockingDelegate(@Nullable WContentBlocking.Delegate delegate) {
+        if (mContentBlockingDelegate == delegate) {
+            return;
+        }
+        mContentBlockingDelegate = delegate;
+        if (mContentBlockingDelegate == null) {
+            mSession.setContentBlockingDelegate(null);
+        } else {
+            mSession.setContentBlockingDelegate(new ContentBlockingDelegateImpl(mContentBlockingDelegate, this));
+        }
+    }
+
+    @Nullable
+    @Override
+    public WContentBlocking.Delegate getContentBlockingDelegate() {
+        return mContentBlockingDelegate;
+    }
+
+    @Override
+    public void setPromptDelegate(@Nullable PromptDelegate delegate) {
+        if (mPromptDelegate == delegate) {
+            return;
+        }
+        mPromptDelegate = delegate;
+        if (mPromptDelegate == null) {
+            mSession.setPromptDelegate(null);
+        } else {
+            mSession.setPromptDelegate(new PromptDelegateImpl(mPromptDelegate, this));
+        }
+    }
+
+    @Nullable
+    @Override
+    public PromptDelegate getPromptDelegate() {
+        return mPromptDelegate;
+    }
+
+    @Override
+    public void setSelectionActionDelegate(@Nullable WSession.SelectionActionDelegate delegate) {
+        if (mSelectionActionDelegate == delegate) {
+            return;
+        }
+        mSelectionActionDelegate = delegate;
+        if (mContentDelegate == null) {
+            mSession.setSelectionActionDelegate(null);
+        } else {
+            mSession.setSelectionActionDelegate(new SelectionActionDelegateImpl(mSelectionActionDelegate, this));
+        }
+    }
+
+    @Override
+    public void setMediaSessionDelegate(@Nullable WMediaSession.Delegate delegate) {
+        if (mMediaSessionDelegate == delegate) {
+            return;
+        }
+        mMediaSessionDelegate = delegate;
+        if (delegate == null) {
+            mSession.setMediaSessionDelegate(null);
+        } else {
+            mSession.setMediaSessionDelegate(new MediaSessionDelegateImpl(this, delegate));
+        }
+    }
+
+    @Nullable
+    @Override
+    public WMediaSession.Delegate getMediaSessionDelegate() {
+        return mMediaSessionDelegate;
+    }
+
+    @Nullable
+    @Override
+    public WSession.SelectionActionDelegate getSelectionActionDelegate() {
+        return mSelectionActionDelegate;
+    }
+
+
+    private int toGeckoFlags(@LoadFlags int flags) {
+        int result = 0;
+        if ((flags & WSession.LOAD_FLAGS_NONE) != 0) {
```

Comment:
Do we really need the `!= 0` ? Same for the other if conditions

Reply:
Java doesn't convert to boolean